core/src/modules/ops/appRefreshQuery.ts

```diff
@@ -31,6 +31,14 @@ export namespace AppRefreshQueryOps {
   }
 
   export async function triggerSchedules(appRefreshQuery: AppRefreshQuery) {
+    //if there are runs that were triggered by this that are still running, don't save anything and don't trigger anything
+    const activeRuns = await Run.findAll({
+      where: { state: "running", triggeredBy: appRefreshQuery.id },
+    });
```

Comment:
You don't need to load up all the runs (findAll), you really just want to see if any are running - try `.count()`